src/protocol-parser/protocol-parser.ts

```diff
@@ -270,24 +270,27 @@ export namespace Script {
   //   ?maxObjectDepth: (js-uint / null) .default null,
   //   ?includeShadowTree: ("none" / "open" / "all") .default "none",
   // }
-  const SerializationOptions = zod.object({
-    maxDomDepth: zod.number().int().min(0).optional(),
-    maxObjectDepth: zod.number().int().min(0).max(MAX_INT).optional(),
+  const SerializationOptionsSchema = zod.object({
+    maxDomDepth: zod.union([zod.null(), zod.number().int().min(0)]).optional(),
```

Comment:
How about adding `.default(0)` here? https://w3c.github.io/webdriver-bidi/#type-script-SerializationOptions

Reply:
If we want to provide a BiDi-like API to Puppeteer, we have to manage defaults later on in `process_script_evaluate` and `process_script_callFunction`. Handling defaults in 2 places seems odd.

Reply:
OK. Then maybe we should change `browsingContext.print` to match and stop using `.default` here. But this is out of scope. Feel free to keep it as is then.